### ISSUE TITLE
Failures reading light entity for Hampton Bay Fan combo

### ISSUE THREAD

**Issue by shawngmc** [jdeath/Hubspace-Homeassistant#16]
I'm trying to add my fan combo for the first time, and it appears to be struggling to read the color temp. Log is attached - MAC/SSIC/LatLon removed, UUIDs replaced to keep relationships.

Main error from log:
```
2022-11-29 00:09:50.916 ERROR (MainThread) [homeassistant.components.light] Error while setting up hubspace platform for light
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 289, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 428, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 692, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 789, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 545, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 585, in _async_write_ha_state
    attr.update(self.state_attributes or {})
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 986, in state_attributes
    color_mode = self._light_internal_color_mode
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 819, in _light_internal_color_mode
    if ColorMode.COLOR_TEMP in supported and self.color_temp_kelvin is not None:
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 868, in color_temp_kelvin
    if self._attr_color_temp_kelvin is None and self.color_temp:
  File "/config/custom_components/hubspace/light.py", line 268, in color_temp
    return _convert_color_temp(self._color_temp)
  File "/config/custom_components/hubspace/light.py", line 65, in _convert_color_temp
    return 1000000 // int(value)
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'

```

[hubspace.log](https://github.com/jdeath/Hubspace-Homeassistant/files/10108666/hubspace.log)

**Comment by shawngmc** [comment]
Mine is the Fanelee: https://www.homedepot.com/p/Hampton-Bay-Fanelee-54-in-White-Color-Changing-Integrated-LED-Brushed-Nickel-Smart-Hubspace-Ceiling-Fan-with-Light-Kit-and-Remote-52133/314459956

A model which I believe (1) you recently tried to fix and (2) appears to be a pain in the butt.

Interestingly, whenever the debug output has a values block, it's showing a value that the update code should be able to parse:

```
                {
                    "functionClass": "color-temperature",
                    "lastUpdateTime": 1669590964986,
                    "value": "4000K"
                },
```

However, self._color_temp only appears to be set in turn_on() and update(). Assuming turn_on() isn't being called, is update() being called while the entity is being created, early enough for this to succeed? (I don't know the HA python API, etc. - just spitballing.)

**Comment by jdeath** [comment]
Maybe @dloveall can help. He added the fanlee colortemp support. You could remove the colormode.colortemp line from your model in light.py to avoid the error.

**Comment by jdeath** [comment]
Thanks for your help. To answer why no config flow, no parsing data, or having a fan implemented as a light is simple: I can’t code. This started a quick fix to get my one Hubspace light working. All other devices were added by looking at other’s logs. I took the easiest path. I am open to anyone who can improve this. A config flow would be great as well as a service call to send arbitrary commands. I just haven’t figured it out.
